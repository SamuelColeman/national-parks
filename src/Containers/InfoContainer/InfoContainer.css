.info_title {
	font-size: 8vh;
	padding: 4vh;
	color: #585F29;
	background: #3C321F;
	margin: 0;
}

.info_btn {
	width: 12%;
  height: 4vh;
  margin: 2vh;
  font-size: 3vh;
  border-radius: 5px;
  background: #3C321F;
  color: #585F29;
  border: 1px solid #3C321F;
}

.info_main {
	background: #AD9A7A;
}

.info_btn:hover {
	background: #585F29;
  color: #AD9A7A;
  border: 1px solid #585F29;
}

.info_container {
	background: #AD9A7A;
	max-height: 100vh;
	overflow: scroll;
}