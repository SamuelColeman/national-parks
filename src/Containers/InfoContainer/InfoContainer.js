import React from 'react';
import InfoCard from '../InfoCard/InfoCard';
import './InfoContainer.css';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export const InfoContainer = ({ parkInfo, infoName, parkId }) => {
	const mappedInfo = parkInfo.map(info => {
		return <InfoCard info={info}/>
	})
	if (mappedInfo.length > 0) {
		return (
			<section>
				<h1>{infoName}</h1>
				<Link to={`/parks/${parkId}`}>
					<button>Back</button>
				</Link>
				<div>{mappedInfo}</div>
			</section>
			)
	} else {
		return <h1>No Information Available</h1>
	}
}

export const mapStateToProps = (state) => ({
  parkInfo: state.parkInfo,
  infoName: state.infoName,
  parkId: state.parkId
})

export default connect(mapStateToProps)(InfoContainer);