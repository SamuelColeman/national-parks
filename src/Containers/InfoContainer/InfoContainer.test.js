import React from 'react';
import { shallow } from 'enzyme';
import { InfoContainer, mapStateToProps } from './InfoContainer';

describe('InfoContainer', () => {
	
	let wrapper;
	let mockParkInfo = [{
    	description: "Biggest visitor center ever.",
			directionsInfo: "Take a left.",
			directionsUrl: "https://www.exampledirections.com",
			id: "11",
			latLong: "{lat:35, lng:83}",
			name: "Visitor Center",
			parkCode: "pant",
			url: "https://www.examplecenter.com"
		}];
		let mockInfoName = 'VisitorCenters';
		let mockParkId = '5D';

	beforeEach(() => {
    wrapper = shallow(<InfoContainer parkInfo={mockParkInfo}
    	infoName={mockInfoName}
    	parkId={mockParkId}
    	/>)
 	})

  it('should match snapshot with correct data passing through', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot with correct data passing through', () => {
  	mockParkInfo = [];
    expect(wrapper).toMatchSnapshot();
  });

  it('map state to props gives the park info in state', () => {
		const mockState = { parkInfo: mockParkInfo};

		const expected = {
			parkInfo: mockState.parkInfo
		};

		const mappedState = mapStateToProps(mockState);

		expect(mappedState).toEqual(expected);
	});

	it('map state to props gives the info name in state', () => {
		const mockState = { infoName: mockInfoName};

		const expected = {
			infoName: mockState.infoName
		};

		const mappedState = mapStateToProps(mockState);

		expect(mappedState).toEqual(expected);
	});

	it('map state to props gives the park id in state', () => {
		const mockState = { parkId: mockParkId};

		const expected = {
			parkId: mockState.parkId
		};

		const mappedState = mapStateToProps(mockState);

		expect(mappedState).toEqual(expected);
	});
});