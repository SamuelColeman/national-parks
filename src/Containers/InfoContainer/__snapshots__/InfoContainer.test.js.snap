// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoContainer should match snapshot with correct data passing through 1`] = `
<section>
  <h1>
    VisitorCenters
  </h1>
  <Link
    to="/parks/5D"
  >
    <button>
      Back
    </button>
  </Link>
  <div>
    <Connect(InfoCard)
      info={
        Object {
          "description": "Biggest visitor center ever.",
          "directionsInfo": "Take a left.",
          "directionsUrl": "https://www.exampledirections.com",
          "id": "11",
          "latLong": "{lat:35, lng:83}",
          "name": "Visitor Center",
          "parkCode": "pant",
          "url": "https://www.examplecenter.com",
        }
      }
    />
  </div>
</section>
`;
